QSB/Events/FullStateMessage.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using QSB.Messaging;
+using UnityEngine.Networking;
+
+namespace QSB.Events
+{
+    class FullStateMessage : NameMessage
+    {
+        public override MessageType MessageType => MessageType.FullState;
+
+        public Dictionary<uint, string> PlayerNames;
+
+        public override void Deserialize(NetworkReader reader)
+        {
+            base.Deserialize(reader);
+            PlayerNames = new Dictionary<uint, string>();
+
+            reader.ReadString().Split(',').ToList().ForEach(pair =>
+            {
+                var splitPair = pair.Split(':');
+                var key = Convert.ToUInt16(splitPair[0]);
+                var value = splitPair[1];
+                PlayerNames[key] = value;
+            });
```

Comment:
How about sending a JSON string instead? Then we can use JSON serialization/deserialization instead of string hacking ;)

Reply:
var obj = JsonConvert.DeserializeObject<SomeClass>(jsonString);
var jsonString = JsonConvert.SerializeObject(obj);

JsonConvert is part of a DLL called Newtonsoft.JSON. A .NET Framework 3.5 compatible version is bundled with OWML.

Reply:
JSON is made to be human readable. which we don't really care about here and adds unnecessary bytes. Do you think it's not a big deal?

Reply:
Sold :)